java/vector/src/test/java/org/apache/arrow/vector/util/TestVectorAppender.java

```diff
@@ -0,0 +1,320 @@
+      delta.allocateNew(5, length2);
+
+      ValueVectorDataPopulator.setVector(target, "a0", "a1", "a2", "a3", "a4", "a5", "a6", "a7", "a8", "a9");
+      ValueVectorDataPopulator.setVector(delta, "a10", "a11", "a12", "a13", "a14");
+
+      VectorAppender appender = new VectorAppender(target);
+      delta.accept(appender, null);
+
+      assertEquals(length1 + length2, target.getValueCount());
+      for (int i = 0; i < target.getValueCount(); i++) {
+        assertEquals("a" + i, new String(target.get(i)));
+      }
+    }
+  }
+
+  @Test
+  public void testAppendListVector() {
+    final int length1 = 5;
+    final int length2 = 2;
+    try (ListVector target = ListVector.empty("target", allocator);
+         ListVector delta = ListVector.empty("delta", allocator)) {
+
+      target.allocateNew();
+      ValueVectorDataPopulator.setVector(target,
+          Arrays.asList(0, 1),
+          Arrays.asList(2, 3),
+          Arrays.asList(4, 5),
+          Arrays.asList(6, 7),
+          Arrays.asList(8, 9));
+      assertEquals(length1, target.getValueCount());
+
+      delta.allocateNew();
+      ValueVectorDataPopulator.setVector(delta,
+          Arrays.asList(10, 11, 12, 13, 14),
+          Arrays.asList(15, 16, 17, 18, 19));
+      assertEquals(length2, delta.getValueCount());
+
+      VectorAppender appender = new VectorAppender(target);
+      delta.accept(appender, null);
+
+      assertEquals(7, target.getValueCount());
+
+      int curValue = 0;
+      for (int i = 0; i < length1; i++) {
+        List<Integer> list = (List<Integer>) target.getObject(i);
+        assertEquals(2, list.size());
+        for (int j = 0; j < list.size(); j++) {
+          assertEquals(curValue++, list.get(j).intValue());
+        }
+      }
+
+      for (int i = length1; i < length1 + length2; i++) {
+        List<Integer> list = (List<Integer>) target.getObject(i);
+        assertEquals(5, list.size());
+        for (int j = 0; j < list.size(); j++) {
+          assertEquals(curValue++, list.get(j).intValue());
+        }
+      }
+    }
+  }
+
+  @Test
+  public void testAppendFixedSizeListVector() {
+    try (FixedSizeListVector target = FixedSizeListVector.empty("target", 5, allocator);
+         FixedSizeListVector delta = FixedSizeListVector.empty("delta", 5, allocator)) {
+
+      target.allocateNew();
+      ValueVectorDataPopulator.setVector(target,
+          Arrays.asList(0, 1, 2, 3, 4),
+          Arrays.asList(5, 6, 7, 8, 9));
+      assertEquals(2, target.getValueCount());
+
+      delta.allocateNew();
+      ValueVectorDataPopulator.setVector(delta,
+          Arrays.asList(10, 11, 12, 13, 14),
+          Arrays.asList(15, 16, 17, 18, 19));
+      assertEquals(2, delta.getValueCount());
+
+      VectorAppender appender = new VectorAppender(target);
+      delta.accept(appender, null);
+
+      assertEquals(4, target.getValueCount());
+      int curValue = 0;
+      for (int i = 0; i < target.getValueCount(); i++) {
```

Comment:
I thought I saw we have an assertVectorsEquals someplace in the code?

Reply:
Sorry I did not find one. So I have created one.